R/fit.R

```diff
@@ -117,45 +109,45 @@ h_summarize_all_fits <- function(all_fits) {
 #' )
 #' best_fit <- refit_multiple_optimizers(fit)
 refit_multiple_optimizers <- function(fit,
-                                      n_cores = 1L,
-                                      optimizers = c("L-BFGS-B", "BFGS", "CG", "nlminb"),
-                                      accept_singular = TRUE) {
+                                      ...,
+                                      control = mmrm_control(...)) {
   assert_class(fit, "mmrm_fit")
-  assert_int(n_cores, lower = 1L)
-  optimizers <- match.arg(optimizers, several.ok = TRUE)
-
+  assert_class(control, "mmrm_control")
   # Extract the components of the original fit.
   old_formula <- formula(fit)
   old_data <- fit$data
   old_weights <- fit$weights
-  old_optimizer <- attr(fit, "optimizer")
 
-  # Settings for the new fits.
-  optimizers <- setdiff(optimizers, old_optimizer)
+
   n_cores_used <- ifelse(
     .Platform$OS.type == "windows",
     1L,
     min(
-      length(optimizers),
-      n_cores
+      length(control$optimizers),
+      control$n_cores
     )
   )
+  # split controls
+  controls <- h_split_control(
+    control,
+    start = fit$theta_est
+  )
 
   # Take the results from old fit as starting values for new fits.
-  all_fits <- suppressWarnings(parallel::mclapply(
-    X = optimizers,
+  all_fits <- suppressWarnings(parallel::mcmapply(
     FUN = fit_single_optimizer,
-    formula = old_formula,
-    data = old_data,
-    weights = old_weights,
-    reml = fit$reml,
-    start = fit$theta_est,
-    accept_singular = accept_singular,
+    control = controls,
+    MoreArgs = list(
+      formula = old_formula,
+      data = old_data,
+      weights = old_weights,
+      reml = fit$reml
+    ),
     mc.cores = n_cores_used,
-    mc.silent = TRUE
+    mc.silent = TRUE,
+    SIMPLIFY = FALSE
   ))
   all_fits <- c(all_fits, list(old_result = fit))
-  names(all_fits) <- c(optimizers, old_optimizer)
```

Comment:
this should still work; but since it does not make much sense in "customized optimizer" things I removed this ( optimizers are now partial functions with some arguments instead of a bare string, having to name them all became a little challenging)